openprescribing/matrixstore/tests/commands/test_matrixstore_build.py

```diff
@@ -0,0 +1,212 @@
+            call_command(
+                'matrixstore_build',
+                end_date,
+                cls.data_file,
+                months=len(cls.months_to_import),
+                quiet=True
+            )
+
+    @classmethod
+    def tearDownClass(cls):
+        shutil.rmtree(cls.tempdir)
+
+    def setUp(self):
+        # We have to check this because the `sqlite3.connect` call will
+        # implicitly create the file if it doesn't exist
+        if not os.path.exists(self.data_file):
+            raise RuntimeError('No SQLite file created')
+        self.connection = sqlite3.connect(self.data_file)
+
+    def tearDown(self):
+        self.connection.close()
+
+    def test_dates_are_correct(self):
+        dates = [
+            row[0] for row in
+            self.connection.execute('SELECT date FROM date ORDER BY offset')
+        ]
+        expected_dates = [date[:10] for date in self.months_to_import]
+        self.assertEqual(dates, expected_dates)
+
+    def test_practices_are_correct(self):
+        practice_codes = [
+            row[0] for row in
+            self.connection.execute('SELECT code FROM practice ORDER BY code')
+        ]
+        expected_codes = sorted(
+            [practice['code'] for practice in self.active_practices]
+        )
+        self.assertEqual(practice_codes, expected_codes)
+        self.assertNotIn(self.closed_practice['code'], practice_codes)
+
+    def test_presentations_are_correct(self):
+        expected = list(self.presentations)
+        expected.append(self.updated_presentation)
+        expected.sort(key=lambda i: i['bnf_code'])
+        # Allow us to get results as dicts
+        self.connection.row_factory = sqlite3.Row
+        results = self.connection.execute(
+            """
+            SELECT
+              bnf_code, is_generic, adq_per_quantity, name
+            FROM
+              presentation
+            ORDER BY
+              bnf_code
+            """
+        )
+        results = [dict(row) for row in results]
+        self.assertEqual(results, expected)
+        self.assertNotIn(self.presentation_to_update, results)
+
+    def test_practice_statistics_are_correct(self):
+        get_value = MatrixValueFetcher(
+            self.connection, 'practice_statistic', 'name', 'value'
+        )
+        for entry in self.practice_statistics:
+            practice = entry['practice']
+            month = entry['month']
+            if month not in self.months_to_import:
+                continue
+            expected_total_list_size = entry['total_list_size']
+            expected_statins_cost = json.loads(entry['star_pu'])['statins_cost']
+            total_list_size = get_value('total_list_size', practice, month)
+            statins_cost = get_value('star_pu.statins_cost', practice, month)
+            self.assertEqual(total_list_size, expected_total_list_size)
+            self.assertEqual(statins_cost, expected_statins_cost)
```

Comment:
We're testing that every expected row is present, but can you can you also test that there are no extra rows in the database?